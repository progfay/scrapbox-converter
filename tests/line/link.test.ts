/* global describe it expect */

import { parse } from '@progfay/scrapbox-parser'
import convert from '../../src'
import pretty from 'pretty'

describe('link', () => {
  it('Simple link', () => {
    const input = `Link
https://example.com`
    const html = convert(parse(input), 'projectName')
    expect(pretty(html)).toMatchSnapshot()
  })

  it('Link with content', () => {
    const input = `Link
[https://example.com example]`
    const html = convert(parse(input), 'projectName')
    expect(pretty(html)).toMatchSnapshot()
  })

  it('Relative link', () => {
    const input = `Link
[settings]`
    const html = convert(parse(input), 'projectName')
    expect(pretty(html)).toMatchSnapshot()
  })

  it('Root link', () => {
    const input = `Link
[/help/syntax]`
    const html = convert(parse(input), 'projectName')
    expect(pretty(html)).toMatchSnapshot()
  })
})
