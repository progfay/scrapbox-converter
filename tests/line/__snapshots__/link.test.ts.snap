// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`link Link with content 1`] = `
"<article>
  <h1 class=\\"title\\">Link</h1>
  <section>
    <p><a href=\\"https://example.com\\">example</a></p>
  </section>
</article>"
`;

exports[`link Relative link 1`] = `
"<article>
  <h1 class=\\"title\\">Link</h1>
  <section>
    <p><a href=\\"settings\\">settings</a></p>
  </section>
</article>"
`;

exports[`link Root link 1`] = `
"<article>
  <h1 class=\\"title\\">Link</h1>
  <section>
    <p><a href=\\"https://scrapbox.io/help/syntax\\">/help/syntax</a></p>
  </section>
</article>"
`;

exports[`link Simple link 1`] = `
"<article>
  <h1 class=\\"title\\">Link</h1>
  <section>
    <p><a href=\\"https://example.com\\">https://example.com</a></p>
  </section>
</article>"
`;
